src/Cody.VisualStudio/Services/EditCodeService.cs

```diff
@@ -33,6 +55,15 @@ public EditCodeResult ShowEditCodeDialog(IEnumerable<EditModel> models, string d
 
             return result;
         }
+
+        private void SaveInstructionInHistory(string instruction)
+        {
+            instructionsHistory.Add(instruction);
+            var itemsToRemove = instructionsHistory.Count - MaxHistoryItems;
+            if (itemsToRemove > 0) instructionsHistory = instructionsHistory.Skip(itemsToRemove).ToList();
```

Comment:
can we mutate the list in-place? smth like:
```
instructionsHistory.RemoveRange(0, itemsToRemove)
```
???

or we do need the new reference?